ipam/calico-ipam.go

```diff
@@ -0,0 +1,91 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"net"
+
+	"github.com/containernetworking/cni/pkg/skel"
+	"github.com/containernetworking/cni/pkg/types"
+	"github.com/projectcalico/libcalico/lib/ipam"
+)
+
+func main() {
+	skel.PluginMain(cmdAdd, cmdDel)
+}
+
+// IPAMConfig represents the IP related network configuration.
+type IPAMConfig struct {
```

Comment:
Making a note: will need to support https://github.com/projectcalico/calico-cni/blob/master/ipam.py#L62